### PR TITLE
Update rmc-docs after recent changes to RMC

### PR DESCRIPTION
**Pull request by celinval**
### Description of changes: 

Update rmc-docs after recent changes to RMC:

  - Better cargo rmc support.
  - Adding rmc-compiler executable.

### Resolved issues:

N/A

### Testing:

* How is this change tested? Yes

* Is this a refactor change?

### Checklist
- [ ] Each commit message has a non-empty body, explaining why the change was made
- [ ] Methods or procedures are documented
- [ ] Regression or unit tests are included, or existing tests cover the modified code
- [ ] My PR is restricted to a single feature or bugfix

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 and MIT licenses.

### CODE REVIEW

File: rmc-docs/src/cargo-rmc.md [hunk 1]
```diff
@@ -1,52 +1,41 @@
...
 
 # Building and running
 
-Let's assume you have a project you can build with `cargo build` and you've written a proof harness somewhere in it that you want to run RMC on:
+Let's assume you have a library project that you can build with `cargo build` and you've written a proof harness somewhere in it that you want to run RMC on:
 
 ```rust
-#[no_mangle]
 #[cfg(rmc)]
+#[rmc::proof]
```
Comment (zhassan-aws) [inline rmc-docs/src/cargo-rmc.md:11]: I think we should add a section before this one that introduces the `#[rmc::proof]` annotation.

File: rmc-docs/src/cargo-rmc.md [hunk 2]
```diff
@@ -1,52 +1,41 @@
...
+For example, in order to configure a maximum loop unwind threshold for harnesses in a package, you can add the following line to your package `Cargo.toml`:
+```toml
+[package.metadata.rmc]
+flags= { unwind = "5" }
 ```
 
-At this point we have the project built, but now we want to transform it into something that will run a specific proof harness.
-To do that, we specialize it, preprocess it, and then run CBMC on the result:
-(In practice, we might want to do the above steps once, then repeat the below steps for each proof harness.)
+Note that RMC will extract any extra configuration from sections `[rmc]`, `[workspace.metadata.rmc]`, `[package.metadata.rmc]` in the TOML file.
```
Comment (zhassan-aws) [inline rmc-docs/src/cargo-rmc.md:31]:
```suggestion
Note that RMC will extract any extra configuration from sections `[rmc]`, `[workspace.metadata.rmc]`, and `[package.metadata.rmc]` in the TOML file.
```

File: rmc-docs/src/cargo-rmc.md [hunk 3]
```diff
@@ -1,52 +1,41 @@
...
 
-```bash
-{{#include sample-rmc-build.sh:cbmc}}
-```
+# Common cargo-rmc arguments
+
+Cargo rmc supports the rmc standalone arguments described in the [RMC single file section](./rmc-single-file.md). Additionally, it also accepts the following arguments:
+
+**`--config-toml`** Location of a configuration file in toml format for your project. This defaults to to crate's Cargo.toml.
```
Comment (zhassan-aws) [inline rmc-docs/src/cargo-rmc.md:37]:
```suggestion
**`--config-toml`** Location of a configuration file in toml format for your project. This defaults to the crate's Cargo.toml.
```

File: rmc-docs/src/cargo-rmc.md (hunk 1 above)
Comment (celinval) [reply rmc-docs/src/cargo-rmc.md:11]: What if I just remove this example. It's not adding much value anymore.

Comment (zhassan-aws) [reply rmc-docs/src/cargo-rmc.md:11]: And only leave the `cargo rmc --function my_harness` command in this section? If so, my understanding is that the command wouldn't work without annotating the function with either `#[no_mangle]` or `#[rmc::proof]`.

**Review by zhassan-aws — APPROVED** [review]

**Pull request merged**